src/contracts/atlas/Atlas.sol

```diff
@@ -220,20 +213,91 @@ contract Atlas is Test, Factory {
             }
         }
 
-        // Only verify signatures of meta txs if the original signer isn't the bundler
-        // TODO: Consider extra reentrancy defense here?
-        if (dAppOp.approval.from != msg.sender && !_verifyDApp(dConfig, dAppOp)) {
-            bool bypass = isSimulation && dAppOp.signature.length == 0;
-            if (!bypass) {
+        // bundler checks
+        // user is bundling their own operation - check allowed and valid dapp signature/callchainhash
+        if(msg.sender == userOp.call.from) {
+            if(!dConfig.callConfig.allowsUserBundler()) {
+                return ValidCallsResult.UnknownBundlerNotAllowed;
+            }
+
+            // user should not include their own signature, they already signed the transaction
+            if(userOp.signature.length > 0) {
```

Comment:
Is there any safety reason to explicitly revert if user includes their sig _and is also_ the bundler?

Might be worth removing this check in the spirit of cutting down code and gas cost